packages/smooth_app/lib/pages/page_manager.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:flutter/material.dart';
+import 'package:smooth_app/pages/scan/scan_page.dart';
+import 'package:smooth_app/widgets/tab_navigator.dart';
+
+enum BottomNavigationTab {
+  Profile,
+  Scan,
+  History,
+}
+
+class PageManager extends StatefulWidget {
```

Comment:
I would add some comments here too, in order to explain the issue. Something like "Stateful tabs management".
Or link to the issue.